proto/ibc/core/port/v1/query.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package ibc.core.port.v1;
+
+option go_package = "github.com/cosmos/ibc-go/modules/core/05-port/types";
+
+import "ibc/core/channel/v1/channel.proto";
+
+// Query provides defines the gRPC querier service
+service Query {
+  // Port queries an IBC Port.
+  rpc Port(QueryPortRequest) returns (QueryPortResponse) {}
+}
+
+// QueryPortRequest is the request type for the Query/Port RPC method
+message QueryPortRequest {
+  // port unique identifier
+  string port_id = 1;
+  // counterparty channel end
+  ibc.core.channel.v1.Counterparty counterparty = 2;
+  // counterparty version
+  string counterparty_version = 3;
```

Comment:
Does the counterparty version need to be included in the request here?

Reply:
Yep, it's currently being used in the `transfer` module's `NegotiateAppVersion` implementation, and I think we would also need it for the ICA approach

Reply:
Yes this is needed for app version negotiation. We could rename it to `proposed_version` as well, but `counterparty_version` works

Reply:
I'm a little confused by this. I thought the point is that at this stage we don't actually know what the `counterparty_version` is?

Reply:
The confusion is what `counterparty` means. This probably means we should change the naming

Happy path:
chainA - ChanOpenInit
chainB - ChanOpenTry

preChanOpenTry, the relayer may need to use this gRPC to determine what version to supply to the channel on chainB (for ChanOpenTry). So if this is called on `chainB` the counterparty is `chainA`. Obviously this could be called on chainA as well. I recommend we use `proposed_version` and title this gRPC `NegotiateAppVersion` - no context of what is the `counterparty` is needed

Reply:
Agreed, already started to make these changes, will push an update and include other suggestions also!

Reply:
makes sense!